### DISCUSSION TITLE
Simple, working Bluetooth with cornrow

### DISCUSSION THREAD

**Discussion by eriqjaffe** [snapcast/snapcast#911] [General]
I have successfully added Bluetooth to snapcast using [cornrow](https://github.com/mincequi/cornrow).  I had been playing around with various tutorials to make a Raspberry Pi into a bluetooth receiver and none of them worked to my satisfaction - but cornrow is just dead simple and works.

By default cornrow will output to ALSA which I've always found a bugger to work with.  The dev just added the ability to output to a TCP/IP stream in the 0.8.0 release which makes it a breeze to add to snapcast.  Per the dev, I modified the /etc/cornrowd.conf file as follows:

```
# This is a TOML document.

[bluetooth_source]
codecs = [ 0x0 ]        # As defined in the A2DP specification (SBC (0x0) is ma$
sample_rates = [ 44100, 48000 ]

#[airplay_source]
#name = "myAirplay"
# port = 0              # port to listen on (0 for auto select)
# buffer_time = 2000    # in ms (defaults to 2000ms)

#[alsa_sink]
#device = "default"

[tcp_sink]
host = "127.0.0.1"
port = 4953
```

And added the following to /etc/snapcast.conf:

`source=tcp://127.0.0.1:4953?name=cornrow&sampleformat=44100:16:2`

...and it just works beautifully.  Might be a good idea to add this to the documentation since it's a very simple solution.

**Comment by uSpike** [comment]
Oh wow that's great!  I've tried to do this with bluez and pulseaudio and all, and it's a real pain.  I will try this out, thanks.
